### PR TITLE
Add support to any AAD authority when authenticating with Azure Management

### PR DESCRIPTION
**Pull request by skofman1**

### CODE REVIEW

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs [hunk 1]
```diff
@@ -39,6 +41,7 @@ public AzureManagementAPIWrapper(IAzureManagementAPIWrapperConfiguration configu
             }
 
             _clientCredential = new ClientCredential(configuration.ClientId, configuration.ClientSecret);
+            _authority = string.Format(CultureInfo.InvariantCulture, AuthorityTemplate, configuration.AadTenant);
```
Comment (joelverhagen) [inline src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Throw if null? Or default to `microsoft.onmicrosoft.com`?

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs [hunk 2]
```diff
@@ -12,11 +13,12 @@ namespace NuGet.Services.AzureManagement
 {
     public class AzureManagementAPIWrapper : IAzureManagementAPIWrapper
     {
-        private const string Authority = "https://login.microsoftonline.com/microsoft.onmicrosoft.com";
+        private const string AuthorityTemplate = "https://login.microsoftonline.com/{0}";
```
Comment (joelverhagen) [inline src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Are there UTs?

Comment (joelverhagen) [inline src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: Would it be more flexible to make the hole Authority URL config? Or will anyone hitting `management.core.windows.net` need to hit `https://login.microsoftonline.com/`?

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs (hunk 1 above)
Comment (skofman1) [reply src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: added

File: src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs (hunk 2 above)
Comment (skofman1) [reply src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: this class doesn't have much logic. It's all about connecting to Azure. There are UTs around parsing the output.

Comment (skofman1) [reply src/NuGet.Services.AzureManagement/AzureManagementAPIWrapper.cs]: login.microsoftonline.com is a constant URL for AAD auth. It's used for SQL AAD as well: https://github.com/NuGet/ServerCommon/blob/master/src/NuGet.Services.Sql/AzureSqlConnectionStringBuilder.cs#L24

**Pull request merged**